File: client/android/BigFoxAndroid/app/src/main/java/vn/com/onesoft/bigfox/io/core/compress/CompressManager.java
package vn.com.onesoft.bigfox.io.core.compress;

/**
 * Created by phamquan on 8/18/15.
 */
public class CompressManager {

    private ICompress cp;

    private static CompressManager _instance = null;

    public static CompressManager getInstance() {
        if (_instance == null) {
            _instance = new CompressManager();
            _instance.setCompress(new BFZip());
        }
        return _instance;
    }

    private void setCompress(ICompress compress){
        this.cp = compress;
    }

    public byte[] compress(byte[] data){
        if(true)
            return data;
        return cp.compress(data);
    }

    public byte[] decompress(byte[] data){
        return cp.decompress(data);
    }
}

File: client/ios/BigFoxClient/BigFoxClient/core/compress/CompressManager.m
//
//  CompressManager.m
//  BigFoxClient
//
//  Created by QuyenNX on 9/4/15.
//  Copyright (c) 2015 QuyenNX. All rights reserved.
//

#import "CompressManager.h"
#import "BFZip.h"

@implementation CompressManager {
    id<ICompress> cp;
}

static CompressManager* _instance = nil;

+ (CompressManager*) getInstance {
    if(_instance == nil) {
        _instance = [[CompressManager alloc] init];
        [_instance setCompress:[[BFZip alloc]init]];
    }
    return _instance;
}

- (void) setCompress : (id<ICompress>) compress {
    cp = compress;
}
-(NSData*) compress:(NSData *)data {
    return [cp compress:data ];
}

-(NSData*) decompress:(NSData *)data{
    return [cp decompress:data ];
}
@end
